ds-and-algo/answers/gopal-q29.html

```diff
@@ -0,0 +1,61 @@
+<html>
+    <head>
+
+        <!-- Global site tag (gtag.js) - Google Analytics -->
+        <script async src="https://www.googletagmanager.com/gtag/js?id=UA-157837769-1"></script>
+        <script>
+        window.dataLayer = window.dataLayer || [];
+        function gtag(){dataLayer.push(arguments);}
+        gtag('js', new Date());
+        gtag('config', 'UA-157837769-1');
+        </script>
+        
+        <title>
+            Data Structures and Algorithms Questions | InterQues | Coding Ninjas
+        </title>
+
+        <link rel="shortcut icon" type="image/png" href="../../assets/favicon.png"/>
+
+        <meta name="description" content="How to check if a given Binary Tree is BST or not?">
+    </head>
+    <body>
+
+        <div>
+            <img src="../../assets/CN_logo.png" height='72px' alt="Coding Ninjas Logo">
+        </div>
+
+        <div>
+            <p>
+                <small>
+
+                    <a href="../../index.html">Home</a>
+                    &gt; <a href="../index.html">DS Questions</a>
+                    &gt; How to check if a given Binary Tree is BST or not?
+
+                </small>
+               
+            </p>
+        </div>
+
+
+        <div>
+            <div>
+                <h4>
+                    How to check if a given Binary Tree is BST or not?
+                </h4>
+            <br>
+            <br>
+            <div>
+                <strong>Answer:</strong> 
+                <p>
+                    If inorder traversal of a binary tree is sorted, then the binary tree is BST. 
+                    The idea is to simply do inorder traversal and while traversing keep track of previous key value. 
+                    If current key value is greater, then continue, else return false.
+                </p>
```

Comment:
Provide the code too.

Reply:
i should send u code or i should send u pull request again?

Reply:
private void inOrder(TreeNode node) {
    if (node == null) {
      return;
    }

    inOrder(node.left);
    System.out.printf("%s ", node.data);
    inOrder(node.right);
}

Reply:
No, just create another commit in this branch locally and then push the changes.